build.gradle

```diff
@@ -1,42 +1,20 @@
-buildscript {
-    repositories {
-        maven { url "https://plugins.gradle.org/m2/" }
-        mavenCentral()
-        google()
-    }
-
-    ext.agpVersion = "7.2.0"
-    ext.kotlinVersion = "1.4.31"
-
-    dependencies {
-        classpath "com.android.tools.build:gradle:$agpVersion"
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion"
-        classpath "io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.17.1"
-        classpath "org.jlleitschuh.gradle:ktlint-gradle:10.1.0"
-    }
-}
-// Gradle plugins
 plugins {
-    id "java-gradle-plugin"
-    id "groovy"
-    id 'maven-publish'
-    id "com.gradle.plugin-publish" version "0.15.0"
-    id "com.vanniktech.maven.publish" version "0.15.1"
-    id "com.github.hierynomus.license" version "0.16.1"
-}
+    id("groovy")
 
-apply plugin: "org.jetbrains.kotlin.jvm"
-apply plugin: "org.jetbrains.kotlin.kapt"
-apply plugin: "io.gitlab.arturbosch.detekt"
-apply plugin: "org.jlleitschuh.gradle.ktlint"
+    id("org.jetbrains.kotlin.kapt") version "1.7.10"
+    id("org.jetbrains.kotlin.jvm") version "1.7.10"
```

Comment:
Should we use `kotlinVersion` here too? Or is it not necessarily the same thing?

Reply:
We can't because the `plugins` block can't reference it directly. We should move to using a version catalogue, but I think that's beyond the scope of this PR?